src/extension/fileSystemProviders/scriptSourceFileSystemProvider.ts

```diff
@@ -35,14 +47,32 @@ export function getScriptSourceLocation(location: Location, sources: Sources | u
 
 export function wrapScriptSource(source: ScriptSource) {
     if (source.scriptId && source.scriptId > 0) {
+        let path = "";
+        const query = new URLSearchParams();
+        query.set("scriptId", `${source.scriptId}`);
+        if (source.uri) {
+            query.set("uri", source.uri.toString(/*skipEncoding*/ false));
+            switch (source.uri.scheme) {
+                case "file":
+                case "vscode-remote":

```

Comment:
FWIW there are others but I can't recall the full list right now... Any extension can invent its own scope and screw with things.

Reply:
This is primarily used to give a better title in the document tab in the editor. For URIs that have paths that behave like actual paths, I use the path itself. For URIs that I can't be sure of, I use the encoded form a few lines below.